pkg/gardenlet/controller/bastion/add.go

```diff
@@ -0,0 +1,57 @@
+// Copyright (c) 2022 SAP SE or an SAP affiliate company. All rights reserved. This file is licensed under the Apache Software License, v. 2 except as noted otherwise in the LICENSE file
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package bastion
+
+import (
+	operationsv1alpha1 "github.com/gardener/gardener/pkg/apis/operations/v1alpha1"
+	"github.com/gardener/gardener/pkg/controllerutils"
+	confighelper "github.com/gardener/gardener/pkg/gardenlet/apis/config/helper"
+
+	"k8s.io/client-go/util/workqueue"
+	"sigs.k8s.io/controller-runtime/pkg/builder"
+	"sigs.k8s.io/controller-runtime/pkg/cluster"
+	"sigs.k8s.io/controller-runtime/pkg/controller"
+	"sigs.k8s.io/controller-runtime/pkg/manager"
+	"sigs.k8s.io/controller-runtime/pkg/predicate"
+)
+
+// ControllerName is the name of this controller.
+const ControllerName = "bastion-controller"
+
+// AddToManager adds Reconciler to the given manager.
+func (r *Reconciler) AddToManager(mgr manager.Manager, gardenCluster, seedCluster cluster.Cluster) error {
+	if r.GardenClient == nil {
+		r.GardenClient = gardenCluster.GetClient()
+	}
+	if r.SeedClient == nil {
+		r.SeedClient = seedCluster.GetClient()
+	}
+
+	return builder.
+		ControllerManagedBy(mgr).
+		For(&operationsv1alpha1.Bastion{}, builder.WithPredicates(r.BastionPredicate(), predicate.GenerationChangedPredicate{})).
+		WithOptions(controller.Options{
+			MaxConcurrentReconciles: *r.Config.Controllers.Bastion.ConcurrentSyncs,
+			RecoverPanic:            true,
+			RateLimiter:             workqueue.DefaultControllerRateLimiter(),
+		}).
+		Complete(r)
```

Comment:
This doesn't work - you cannot use the builder here since you have to overwrite the event handler. You are now trying to watch `operations.gardener.cloud/v1alpha1.Bastion`s in the seed cluster, but those resources are in the garden cluster.